### ISSUE TITLE
IfBuildProfile doesn't work properly with messaging

### ISSUE THREAD

**Issue by ctron** [quarkusio/quarkus#29837]
### Describe the bug

Create a bean which uses e.g. `@Incoming` and annotate it with `@IfBuildProfile`. Then run it without the build profile.

Quarkus now complains about missing configuration if the channels, and still activates them.

### Expected behavior

The bean should be completely disabled. No  warnings, no MQTT connection.

### Actual behavior

The bean is somewhat active. Prices (from the example) don't get generated, but the MQTT connection is active.

### How to Reproduce?

Take a look at the reproducer: https://github.com/ctron/quarkus-repro-1

In a nutshell:

* Annotate all beans with `@IfBuildProfile("foo")`
* Then run without the profile `foo`

### Output of `uname -a` or `ver`

Linux xxx 6.0.12-300.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Dec 8 16:58:47 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18

### Quarkus version or git rev

2.14.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

**Comment by cescoffier** [comment]
Do you mean that the reactive messaging graph is incomplete? That should be considered as a failure and be reported.

**Comment by geoand** [comment]
With 

```java
@IfBuildProfile("foo")
@ApplicationScoped
public class PriceConverter {

    private static final double CONVERSION_RATE = 0.88;

    @Incoming("prices")
    @Outgoing("my-data-stream")
    @Broadcast
    public double process(byte[] priceRaw) {
        int priceInUsd = Integer.parseInt(new String(priceRaw));
        System.out.println("Receiving price: " + priceInUsd);
        return priceInUsd * CONVERSION_RATE;
    }

}
```

and 

```properties
# Configure the MQTT sink (we write to it)
mp.messaging.outgoing.topic-price.type=smallrye-mqtt
mp.messaging.outgoing.topic-price.topic=prices
mp.messaging.outgoing.topic-price.host=localhost
mp.messaging.outgoing.topic-price.port=1883
mp.messaging.outgoing.topic-price.auto-generated-client-id=true

# Configure the MQTT source (we read from it)
mp.messaging.incoming.prices.type=smallrye-mqtt
mp.messaging.incoming.prices.topic=prices
mp.messaging.incoming.prices.host=localhost
mp.messaging.incoming.prices.port=1883
mp.messaging.incoming.prices.auto-generated-client-id=true
```

at runtime we get:

```
2022-12-13 17:56:19,751 WARN  [io.sma.rea.mes.provider] (Quarkus Main Thread) SRMSG00207: Some components are not connected to either downstream consumers or upstream producers:
	- IncomingConnector{channel:'prices', attribute:'mp.messaging.incoming.prices'} has no downstream
	- OutgoingConnector{channel:'topic-price', attribute:'mp.messaging.outgoing.topic-price'} has no upstream

2022-12-13 17:56:19,770 ERROR [io.ver.mqt.imp.MqttClientImpl] (vert.x-eventloop-thread-0) Can't connect to localhost:1883: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:1883
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1589)
```

**Comment by geoand** [comment]
But I still need to check something

**Comment by ozangunalp** [comment]
Shouldn't you also have channel config conditional to that profile?

**Comment by geoand** [comment]
So with the configuration above, the application builds fine. However when one runs it, it fails with:

```
SRMSG00230: Unable to create the publisher or subscriber during initialization: java.lang.IllegalArgumentException: The attribute `host` on connector 'smallrye-mqtt' (channel: my-data-stream) must be set

```

**Comment by geoand** [comment]
> Shouldn't you also have channel config conditional to that profile?

So I guess what we are saying here is that the `my-data-stream` channel is now configured a normal input channel?

**Comment by geoand** [comment]
I think this is looking more like a documentation (and perhaps error message) issue.

**Comment by ozangunalp** [comment]
What I meant is if these config lines are there, without profiles, the reactive messaging will create channels (`prices` incoming, and `topic-price` outgoing) and try to resolve the graph. It obviously can't, because all beans to be removed with `@IfBuildProfile`. 
If a bean using reactive messaging channels is conditional to the build profile, channel config should be conditional too, as  in:
```
%foo.mp.messaging.outgoing.topic-price...
...
%foo.mp.messaging.incoming.prices...
```
Then both `quarkus dev` and `quarkus dev -Dquarkus.profile=foo` would work as expected.

BTW for configuring the connector it should be `connector=smallrye-mqtt` not `type`. (It works for your case because quarkus discovers that only mqtt is available and configures it for you).

**Comment by ctron** [comment]
I can confirm that adding the profile to the properties (`%foo.`) makes it work. Yes that should absolutely be documented. 

But it also feels kind of inconsistent:

* The annotations (e.g `@Incoming`) are part of the bean.
* As is the `@IfBuildProfile` annotation, but it covers the whole bean.

So, disabling a bean should IMHO also disable the messaging stuff.

**Comment by ctron** [comment]
> BTW for configuring the connector it should be `connector=smallrye-mqtt` not `type`. (It works for your case because quarkus discovers that only mqtt is available and configures it for you).

Thanks for the tip, I guess this should be fixed then in the quickstart repo:

https://github.com/quarkusio/quarkus-quickstarts/blob/496cb69159c59fad207c6bd8b82e4a4667dffc09/mqtt-quickstart/src/main/resources/application.properties#L3-L15

**Comment by cescoffier** [comment]
@ctron but the quickstart does not use `@IfBuildProfile`. I think, we should just document that in the reactive messaging reference guide.

**Comment by ctron** [comment]
> @ctron but the quickstart does not use `@IfBuildProfile`. I think, we should just document that in the reactive messaging reference guide.

The quickstart doesn't have the issue reported here. But suffers from an additional issue, as mentioned by @ozangunalp: 

> BTW for configuring the connector it should be `connector=smallrye-mqtt` not `type`. (It works for your case because quarkus discovers that only mqtt is available and configures it for you).

**Comment by cescoffier** [comment]
Ok, so, what is required is just a bit of documentation explaining that the channel configuration must also be prefixed.
